File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/dto/CartItemDto.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.entity.CartItems;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.Product;

import java.util.UUID;

public record CartItemDto(
        UUID cartItemId,
        UUID cartId,
        UUID productId,
        UUID sellerId,
        String sellerName,
        String productName,
        Integer productPrice,
        String categoryName,
        Integer quantity,
        String productImageUrl) {

    public static CartItemDto fromEntity(CartItems cartItems) {
        var cart = cartItems.getCart();
        var product = cartItems.getProduct();
        var seller = product.getSeller();

        return new CartItemDto(
                cartItems.getCartItemId(),
                cart.getCartId(),
                product.getProductId(),
                seller.getSellerId(),
                seller.getSellerName(),
                product.getProductName(),
                product.getProductPrice(),
                product.getCategory().getCategoryName(),
                cartItems.getQuantity(),
                product.getProductImages().isEmpty()
                        ? null : product.getProductImages().getFirst().getImageUrl()
        );
    }

    public static CartItemDto of(Product product, int quantity) {
        var seller = product.getSeller();

        return new CartItemDto(
                null,
                null,
                product.getProductId(),
                seller.getSellerId(),
                seller.getSellerName(),
                product.getProductName(),
                product.getProductPrice(),
                product.getCategory().getCategoryName(),
                quantity,
                product.getProductImages().isEmpty()
                        ? null : product.getProductImages().getFirst().getImageUrl()
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/dto/CartItemDto.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.Product
import java.util.*

data class CartItemDto(
    val cartItemId: UUID?,
    val cartId: UUID?,
    val productId: UUID,
    val sellerId: UUID,
    val sellerName: String,
    val productName: String,
    val productPrice: Int,
    val categoryName: String,
    val quantity: Int,
    val productImageUrl: String?
) {
    companion object {
        @JvmStatic
        fun of(product: Product, quantity: Int): CartItemDto {
            val seller = product.seller

            return CartItemDto(
                null,
                null,
                product.productId,
                seller.sellerId,
                seller.sellerName,
                product.productName,
                product.productPrice,
                product.category.categoryName,
                quantity,
                if (product.productImages.isEmpty())
                    null
                else
                    product.productImages.first().imageUrl
            )
        }
    }
}